src/lib/pci_virtio_sock.c

```diff
@@ -1723,6 +1724,31 @@ static void apply_socket_config_set_buf_alloc(struct pci_vtsock_sock *s,
 	fprintf(stderr, "config: %s buf_alloc = %d\n", s->name, buf_alloc);
 }
 
+static void apply_socket_config_set_name(struct pci_vtsock_sock *s,
+					 struct vtsock_config_hdr *hdr)
+{
+	char *new_name;
+	char *msg_name = (char *)(hdr + 1);
+	size_t name_len = hdr->len - sizeof(struct vtsock_config_hdr);
+
+	name_len = strnlen(msg_name, name_len);
+
+	new_name = malloc(name_len + 22); /* max int string length + : + \0 */
```

Comment:
Oh, I see, 20 is the maximum length of the `%d` formatting of an integer (on this platform at least), which is hard to express as a constant expression. I also see now that this is to be appended to the provided name, hence `s->name` can't be a static array.

Might all this be more clearly expressed as
```
new_thing = asprintf("%*s:%d", msg_len, msg_name, s->id)
free(s->name)
s->name = new_thing
```
??